### PR TITLE
Feature/update about window

### PR DESCRIPTION
**Pull request by BorjaSasieta**
and resolve a bug generated for this Feature.

that is done!

### CODE REVIEW

File: Source/ModuleFontLoader.cpp [hunk 1]
```diff
@@ -13,7 +13,8 @@
 //fonts lib
 #include <ft2build.h>
 #include "MathGeoLib/include/Math/float4x4.h"
-#include FT_FREETYPE_H 
+#include "ft2build.h"
```
Comment (AlbertVVila) [inline Source/ModuleFontLoader.cpp]: duplicated

File: Source/PanelAbout.cpp [hunk 2]
```diff
@@ -8,11 +8,15 @@
 #include "imgui.h"
 #include "rapidjson/rapidjson.h"
 #include "physfs.h"
+#include "freeType/freetype.h"
+#include "mmgr/mmgr.h"
```
Comment (AlbertVVila) [inline Source/PanelAbout.cpp]: Not used anymore

File: Source/PanelAbout.cpp [hunk 3]
```diff
@@ -8,11 +8,15 @@
...
+#include "imguizmo/ImGuizmo.h"
 
 #define ENGINE_DESCRIPTION "Engine created during the Master of Videogames in UPC School."
 #define AUTHOR "Albert Val Vila"
 #define REPOSITORY "https://github.com/AlbertVVila/Engine"
 #define LICENSE "https://github.com/AlbertVVila/Engine/blob/master/LICENSE"
+#define FREETYPE "https://www.freetype.org/"
```
Comment (AlbertVVila) [inline Source/PanelAbout.cpp]: No need to show the website of freetype

File: Source/PanelAbout.cpp [hunk 4]
```diff
@@ -57,14 +61,20 @@ void PanelAbout::Draw()
 		ImGui::BulletText("Assimp (v%d.%d.%d)", aiGetVersionMajor(), aiGetVersionMinor(), aiGetVersionRevision());
 		ImGui::BulletText("Brofiler (v1.1.2)");
 		ImGui::BulletText("DevIL (v%d)", IL_VERSION);
+		if (ImGui::SmallButton(freeTypeVersion))
```
Comment (AlbertVVila) [inline Source/PanelAbout.cpp]: No need to create a button, we only want to show the used version

File: Source/PanelAbout.cpp [hunk 5]
```diff
@@ -57,14 +61,20 @@ void PanelAbout::Draw()
...
+		if (ImGui::SmallButton(freeTypeVersion)) 
+		{
+			ShellExecuteA(NULL, "open", FREETYPE, NULL, NULL, SW_SHOWNORMAL);
+		}
 		ImGui::BulletText("glew (v%s)", glewGetString(GLEW_VERSION));
 		ImGui::BulletText("Imgui (v%s)", ImGui::GetVersion());
 		ImGui::BulletText("Imguizmo");
 		ImGui::BulletText("MathGeoLib (v1.5)");
-		ImGui::BulletText("ParShapes");
+		ImGui::BulletText("mmgr");
```
Comment (AlbertVVila) [inline Source/PanelAbout.cpp]: Library not used anymore

File: Source/PanelAbout.cpp [hunk 6]
```diff
@@ -57,14 +61,20 @@ void PanelAbout::Draw()
 		ImGui::BulletText("Assimp (v%d.%d.%d)", aiGetVersionMajor(), aiGetVersionMinor(), aiGetVersionRevision());
 		ImGui::BulletText("Brofiler (v1.1.2)");
 		ImGui::BulletText("DevIL (v%d)", IL_VERSION);
+		if (ImGui::SmallButton(freeTypeVersion)) 
+		{
+			ShellExecuteA(NULL, "open", FREETYPE, NULL, NULL, SW_SHOWNORMAL);
+		}
 		ImGui::BulletText("glew (v%s)", glewGetString(GLEW_VERSION));
 		ImGui::BulletText("Imgui (v%s)", ImGui::GetVersion());
 		ImGui::BulletText("Imguizmo");
```
Comment (AlbertVVila) [inline Source/PanelAbout.cpp]: Try to find its version

File: Source/PanelAbout.cpp [hunk 7]
```diff
@@ -57,14 +61,20 @@ void PanelAbout::Draw()
...
 		ImGui::BulletText("MathGeoLib (v1.5)");
-		ImGui::BulletText("ParShapes");
+		ImGui::BulletText("mmgr");
+		ImGui::BulletText("NodeEditor");
 		ImGui::BulletText("Pcg (v0.98)");
 		ImGui::BulletText("Physfs (v%d.%d.%d)", PHYSFS_VER_MAJOR, PHYSFS_VER_MINOR, PHYSFS_VER_PATCH);
 		ImGui::BulletText("RapidJson (v%d)", RAPIDJSON_VERSION_STRING);
+		ImGui::BulletText("Recast");
```
Comment (AlbertVVila) [inline Source/PanelAbout.cpp]: Look for the library version

File: Source/PanelAbout.h [hunk 8]
```diff
@@ -2,6 +2,9 @@
 #define __PanelAbout_h__
 
 #include "Panel.h"
+#include "ft2build.h"
```
Comment (AlbertVVila) [inline Source/PanelAbout.h]: Don't include freetype libraries in the PanelAbout.h

File: Source/PanelAbout.h [hunk 9]
```diff
@@ -11,6 +14,9 @@ class PanelAbout : public Panel
 
 	void Draw();
 
+public:
+	std::string toChar = "FreeType (" + std::to_string(FREETYPE_MAJOR)+ "." + std::to_string(FREETYPE_MINOR) + "." + std::to_string(FREETYPE_PATCH) + ")";
```
Comment (AlbertVVila) [inline Source/PanelAbout.h]: No need to do this, do it directly in the .cpp

File: Source/PanelAbout.h [hunk 10]
```diff
@@ -11,6 +14,9 @@ class PanelAbout : public Panel
...
+	const char * freeTypeVersion = toChar.c_str();
```
Comment (AlbertVVila) [inline Source/PanelAbout.h]: Idem

File: Source/PanelAbout.h [hunk 11]
```diff
@@ -11,6 +10,8 @@ class PanelAbout : public Panel
 
 	void Draw();
 
+public:
+	const char * freeTypeVersion;
```
Comment (AlfonsoMestres) [inline Source/PanelAbout.h]: Coding style:   const char* freeTypeVersion  instead const char * freeTypeVersion

**Comment by AlfonsoMestres** [comment]
> and resolve a bug generated for this Feature.
> 
> that is done!

We need something more specific written in there, you should guide the QA to what have you done. For example: 

- Go to Panels
- About window
A new line regarding FreeType should be appearing and then clicked it should go to the page.

**Comment by BorjaSasieta** [comment]
Try now!

File: Source/PanelAbout.h [hunk 12]
```diff
@@ -11,6 +10,8 @@ class PanelAbout : public Panel
 
 	void Draw();
 
+public:
+	const char* freeTypeVersion;
```
Comment (AlbertVVila) [inline Source/PanelAbout.h]: No need to save a pointer for the freetype version

File: Source/PanelAbout.cpp [hunk 13]
```diff
@@ -57,14 +69,20 @@ void PanelAbout::Draw()
 		ImGui::BulletText("Assimp (v%d.%d.%d)", aiGetVersionMajor(), aiGetVersionMinor(), aiGetVersionRevision());
 		ImGui::BulletText("Brofiler (v1.1.2)");
 		ImGui::BulletText("DevIL (v%d)", IL_VERSION);
+		ImGui::BulletText(freeTypeVersion);
+		if (ImGui::SmallButton("Click un here to navigate in FreeType's home page")) 
+		{
+			ShellExecuteA(NULL, "open", FREETYPE, NULL, NULL, SW_SHOWNORMAL);
```
Comment (AlbertVVila) [inline Source/PanelAbout.cpp]: I told you, we don't need a button to freetype webpage

File: Source/PanelAbout.cpp [hunk 14]
```diff
@@ -8,11 +8,18 @@
...
+#include "freetype/freetype.h"
 
 #define ENGINE_DESCRIPTION "Engine created during the Master of Videogames in UPC School."
-#define AUTHOR "Albert Val Vila"
-#define REPOSITORY "https://github.com/AlbertVVila/Engine"
-#define LICENSE "https://github.com/AlbertVVila/Engine/blob/master/LICENSE"
+#define AUTHOR "FRACTAL PUPPY"
+#define REPOSITORY "https://github.com/fractal-puppy/Engine"
+#define LICENSE "https://github.com/fractal-puppy/Engine/blob/master/LICENSE"
+#define FREETYPE "https://www.freetype.org/"
```
Comment (AlbertVVila) [inline Source/PanelAbout.cpp]: not needed

**Comment by AlfonsoMestres** [comment]
image

We do not need a special link to freetype, it's not any special library. Remove the link "click un here.." please.

**Comment by BorjaSasieta** [comment]
now it's ok!

File: Source/PanelAbout.cpp [hunk 15]
```diff
@@ -32,6 +38,10 @@ void PanelAbout::Draw()
 		return;
 	}
 
+	std::string toChar = "FreeType (";
+	toChar = toChar + std::to_string(FREETYPE_MAJOR) + "." + std::to_string(FREETYPE_MINOR) + "." + std::to_string(FREETYPE_PATCH) + ")";
+	freeTypeVersion = toChar.c_str();
```
Comment (RogerNogue) [inline Source/PanelAbout.cpp]: Everything looks good except the method you are using with the FreeType version. You should not use a pointer to char and update it every frame only for the FreeType version

**Comment by BorjaSasieta** [comment]
Done!

**Comment by Gabroide** [comment]
Reviewed

**Pull request merged**